src/main/scala/com/microsoft/hyperspace/index/rankers/FilterIndexRanker.scala

```diff
@@ -54,7 +54,12 @@ object FilterIndexRanker {
         } else {
           // TODO: Add ranking algorithm to sort candidates.
           //  See https://github.com/microsoft/hyperspace/issues/52
-          Some(candidates.head)
+
+          // Pick the index with minimum size. If indexes with same size are found, pick the
+          // one with lexicographically smaller name. This is required for deterministic selection
+          // of indexes.
+          Some(candidates.minBy(index =>
+            (index.getTagValue(plan, IndexLogEntryTags.INDEX_SIZE_IN_BYTES), index.name)))
```

Comment:
Could you use `sourceFilesSizeInBytes` in IndexLogEntry?

Reply:
i think that is for source data, not index data

Reply:
pl. let me know if i am wrong

Reply:
ok.. then can we just add `lazy val indexFilesSizeInBytes` instead of tag?
since I think we can use this condition for join index ranker.